webpack/assets/javascripts/react_app/components/ForemanModal/subcomponents/SubmitOrCancel/SubmitOrCancelActions.js

```diff
@@ -0,0 +1,53 @@
+import { API } from '../../../../redux/API';
+
+import { addToast } from '../../../../redux/actions/toasts';
+import { translate as __ } from '../../../../common/I18n';
+
+const onModalError = error => {
+  const { response } = error;
+
+  if (response.data && response.data.error) {
+    if (response.data.error.message) {
+      return response.data.error.message;
+    }
+
+    if (response.data.error.full_messages) {
+      return response.data.error.full_messages.join(', ');
+    }
+  }
+
+  return `${error.response.status}: ${__('Failed to submit the request.')}`;
+};
+
+export const submitModal = ({
+  url,
+  message,
+  method = 'delete',
+  closeFn,
+  onErr = onModalError,
+  onSucc = () => {},
```

Comment:
I'm not sucking, I'm experiencing success.  Prefer the unabbreviated names `onSuccess` and `onError` :smiley: